app/src/main/java/ryey/easer/skills/operation/http_request/HttpRequestOperationData.java

```diff
@@ -0,0 +1,194 @@
+/*
+ * Copyright (c) 2016 - 2019 Rui Zhao <renyuneyun@gmail.com>
+ *
+ * This file is part of Easer.
+ *
+ * Easer is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * Easer is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with Easer.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+package ryey.easer.skills.operation.http_request;
+
+import android.os.Build;
+import android.os.Parcel;
+
+import org.jetbrains.annotations.NotNull;
+import org.json.JSONException;
+import org.json.JSONObject;
+
+import java.io.Serializable;
+import java.util.Objects;
+import java.util.Set;
+
+import androidx.annotation.NonNull;
+import androidx.annotation.Nullable;
+
+import ryey.easer.Utils;
+import ryey.easer.commons.local_skill.IllegalStorageDataException;
+import ryey.easer.commons.local_skill.dynamics.SolidDynamicsAssignment;
+import ryey.easer.commons.local_skill.operationskill.OperationData;
+import ryey.easer.plugin.PluginDataFormat;
+
+public class HttpRequestOperationData implements OperationData, Serializable {
+
+    private static final String K_REQUEST_METHOD = "requestMethod";
+    private static final String K_URL = "url";
+    private static final String K_REQUEST_HEADER = "requestHeader";
+    private static final String K_CONTENT_TYPE = "contentType";
+    private static final String K_POST_DATA = "postData";
+
+    @Nullable
+    @Override
+    public Set<String> placeholders() {
+        return null;
+    }
+
+    @NonNull
+    @Override
+    public OperationData applyDynamics(SolidDynamicsAssignment dynamicsAssignment) {
+        return this;
+    }
+
+    enum RequestMethod {
+        GET,
+        POST,
+    }
+
+    @NonNull
+    RequestMethod requestMethod = RequestMethod.GET;
```

Comment:
Would you mind making these fields `final`? (I admit I forgot to do this for network transmission Operation.)

Reply:
Fields of HttpRequestOperationData cannot be declared as `final` since they are written in the parse method.

Reply:
Actually the parse method is only used in the constructor, so it's unnecessary. (There was a historic reason, but no longer needed.)
Never mind, I can change this together with the `placeholders()` things.

Reply:
Oh, I didn't see that. I made them final now.